### ISSUE TITLE
finetune does not work with new lerobot v3 dataset format

### ISSUE THREAD

**Issue by abhishekloiwal** [NVIDIA/Isaac-GR00T#369]
### 🐛 Describe the bug

the finetuning does not work with the new lerobot dataset format (v3) 

### Versions

 python --version && pip freeze

Python 3.12.11
absl-py==2.3.1
aiohappyeyeballs==2.6.1
aiohttp==3.12.15
aiosignal==1.4.0
annotated-types==0.7.0
asttokens==3.0.0
attrs==25.3.0
av==15.1.0
black==25.1.0
certifi==2025.8.3
cffi==1.17.1
charset-normalizer==3.4.3
click==8.2.1
cloudpickle==3.1.1
coacd==1.0.7
colorama==0.4.6
contourpy==1.3.3
cycler==0.12.1
dataclasses-json==0.6.7
datasets==4.1.0
decorator==5.2.1
deepdiff==8.6.1
Deprecated==1.2.18
dill==0.4.0
draccus==0.11.5
etils==1.13.0
execnet==2.1.1
executing==2.2.0
Farama-Notifications==0.0.4
fast_simplification==0.1.12
filelock==3.19.1
fonttools==4.59.2
freetype-py==2.5.1
frozenlist==1.7.0
fsspec==2025.7.0
-e git+https://github.com/Genesis-Embodied-AI/Genesis.git@883129b3592e92ff995cfa256b2cca2bd7bfdf6a#egg=genesis_world
gitdb==4.0.12
GitPython==3.1.45
glfw==2.9.0
grpcio==1.74.0
gstaichi==2.1.1
gymnasium==1.2.0
hf-xet==1.1.9
huggingface-hub==0.34.4
idna==3.10
imageio==2.37.0
imageio-ffmpeg==0.6.0
importlib_resources==6.5.2
iniconfig==2.1.0
inquirerpy==0.3.4
ipython==9.4.0
ipython_pygments_lexers==1.1.1
jedi==0.19.2
Jinja2==3.1.6
kiwisolver==1.4.9
lazy_loader==0.4
libigl==2.6.1
llvmlite==0.44.0
Markdown==3.8.2
markdown-it-py==4.0.0
MarkupSafe==3.0.2
marshmallow==3.26.1
matplotlib==3.10.5
matplotlib-inline==0.1.7
mdurl==0.1.2
mergedeep==1.3.4
ml_dtypes==0.5.3
moviepy==2.2.1
mpmath==1.3.0
mujoco==3.3.5
multidict==6.6.4
multiprocess==0.70.16
mypy_extensions==1.1.0
networkx==3.5
numba==0.61.2
numpy==2.2.6
onnx==1.19.0
opencv-python==4.12.0.88
OpenEXR==3.3.5
orderly-set==5.5.0
packaging==25.0
pandas==2.3.2
parso==0.8.5
pathspec==0.12.1
pexpect==4.9.0
pfzy==0.3.4
pillow==11.3.0
platformdirs==4.4.0
plotly==5.24.1
pluggy==1.6.0
pooch==1.8.2
proglog==0.1.12
prompt_toolkit==3.0.52
propcache==0.3.2
protobuf==6.32.0
psutil==7.0.0
ptyprocess==0.7.0
pure_eval==0.2.3
py==1.11.0
py-cpuinfo==9.0.0
pyarrow==21.0.0
pycollada==0.9.2
pycparser==2.22
pydantic==2.11.7
pydantic_core==2.33.2
PyGEL3D==0.6.1
pyglet==2.1.6
pygltflib==1.16.0
Pygments==2.19.2
pymeshlab==2025.7
pynput==1.8.1
pyobjc-core==11.1
pyobjc-framework-ApplicationServices==11.1
pyobjc-framework-Cocoa==11.1
pyobjc-framework-CoreText==11.1
pyobjc-framework-Quartz==11.1
PyOpenGL==3.1.10
pyparsing==3.2.3
pyserial==3.5
pysplashsurf==0.12.0.2
pytest==8.4.1
pytest-forked==1.6.0
pytest-print==1.1.0
pytest-random-order==1.2.0
pytest-rerunfailures==15.1
pytest-xdist==3.8.0
python-dateutil==2.9.0.post0
python-dotenv==1.1.1
pytz==2025.2
pyvista==0.46.3
PyYAML==6.0.2
pyyaml-include==1.4.1
requests==2.32.5
rich==14.1.0
rsl-rl-lib==2.2.4
rtree==1.4.1
safetensors==0.6.2
scikit-image==0.25.2
scipy==1.16.1
scooby==0.10.1
sentry-sdk==2.35.1
setuptools==80.9.0
six==1.17.0
smmap==5.0.2
stack-data==0.6.3
sympy==1.14.0
syrupy==4.9.1
tenacity==9.1.2
tensorboard==2.20.0
tensorboard-data-server==0.7.2
termcolor==3.1.0
tetgen==0.6.7
tifffile==2025.8.28
toml==0.10.2
torch==2.8.0
torchaudio==2.8.0
torchvision==0.23.0
tqdm==4.67.1
traitlets==5.14.3
trimesh==4.7.4
typing-inspect==0.9.0
typing-inspection==0.4.1
typing_extensions==4.15.0
tzdata==2025.2
urllib3==2.5.0
usd-core==25.8
vtk==9.5.0
wandb==0.21.1
wcwidth==0.2.13
Werkzeug==3.1.3
wrapt==1.17.3
xxhash==3.5.0
yarl==1.20.1
z3-solver==4.15.3.0
zipp==3.23.0
((venv) ) (base) abhishekloiwal@mac genesis-self %

**Comment by zujian-y** [comment]
same issue . which file should we edit to support that fine_tune process?

**Comment by andrearosasco** [comment]
[this](https://github.com/NVIDIA/Isaac-GR00T/blob/main/gr00t/data/dataset.py) I guess. New LerobotDataset classes should be implemented to support V3. Also a conversion script for the dataset would do the work.

**Comment by mohdammari97** [comment]
https://github.com/NVIDIA/Isaac-GR00T/blob/main/getting_started/LeRobot_compatible_data_schema.md

They mention here that GR00T supports v2. I suppose you could check if you could convert your dataset back to v2 with another conversion script. That way you can fine-tune GR00T with your data without problems hopefully.

**Comment by bingogome** [comment]
I submitted two MRs to lerobot and any4lerobot repos. [One](https://github.com/huggingface/lerobot/pull/2109) for v3.0 to v2.1 based off lerobot's v2.1 to v3.0, the [other](https://github.com/Tavish9/any4lerobot/pull/62) for v2.1 to v2.0.

I tested them on my data, and they seem working fine. I am setting GR00T up and will try these converted data on it.
